oak_functions_containers_app/src/lib.rs

```diff
@@ -212,12 +213,49 @@ impl<G: AsyncRecipientContextGenerator + Send + Sync + 'static> OakFunctions
     }
 }
 
+fn create_trace<Body>(request: &http::Request<Body>) -> Span {
+    let uri = request.uri();
+    let mut parts = uri.path().rsplitn(3, "/");
+    let method = parts.next();
+    let service = parts.next();
+
+    // See https://opentelemetry.io/docs/specs/semconv/rpc/rpc-spans/ and
+    // https://opentelemetry.io/docs/specs/semconv/rpc/grpc/ for specifications on what OpenTelemetry
+    // expects the traces to look like. Unfortunately the OTel conventions say that the span name
+    // must be the full RPC method name, but Rust tracing wants the name to be static, so we'll
+    // need to figure something out in the future.
+    tracing::info_span!(
+        "request",
+        rpc.method = method,
+        rpc.service = service,
+        rpc.system = "grpc",
+        rpc.grpc.status_code = tracing::field::Empty,
+        server.address = uri.host(),
+        server.port = uri.port_u16()
+    )
+}
+
+static GRPC_SUCCESS: http::header::HeaderValue = http::header::HeaderValue::from_static("0");
+
 // Starts up and serves an OakFunctionsContainersService instance from the provided TCP listener.
 pub async fn serve<G: AsyncRecipientContextGenerator + Send + Sync + 'static>(
     listener: TcpListener,
     encryption_context: Arc<G>,
 ) -> anyhow::Result<()> {
     tonic::transport::Server::builder()
+        .layer(
+            tower_http::trace::TraceLayer::new_for_grpc()
+                .make_span_with(create_trace)
+                .on_response(|response: &http::Response<_>, _latency, span: &Span| {
+                    let code = response
+                        .headers()
+                        .get("grpc-status")
+                        .unwrap_or(&GRPC_SUCCESS)
```

Comment:
Wouldn't it be safer to fill it in with a generic error if not found, instead of success? e.g. https://docs.rs/tonic/latest/tonic/enum.Code.html#variant.Unknown . Or add a comment explaining why it should be this way.

Reply:
It would be useful to have a comment confirming that, for future reference.